include/ignition/gazebo/components/LightCmd.hh

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#ifndef IGNITION_GAZEBO_COMPONENTS_LIGHTCMD_HH_
+#define IGNITION_GAZEBO_COMPONENTS_LIGHTCMD_HH_
+
+#include <sdf/Light.hh>
+
+#include <ignition/gazebo/config.hh>
+#include <ignition/gazebo/Export.hh>
+
+#include <ignition/gazebo/components/Factory.hh>
+#include "ignition/gazebo/components/Component.hh"
+#include <ignition/gazebo/components/Serialization.hh>
+
+namespace ignition
+{
+namespace gazebo
+{
+// Inline bracket to help doxygen filtering.
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+namespace serializers
+{
+  using LightCmdSerializer =
+      serializers::ComponentToMsgSerializer<sdf::Light, msgs::Light>;
+}
+
+namespace components
+{
+  /// \brief A component type that contains commanded light of an
+  /// entity in the world frame represented by sdf::Light.
+  using LightCmd =
+    Component<sdf::Light, class LightCmdTag, serializers::LightCmdSerializer>;
```

Comment:
I'd recommend using `msgs::Light` as the data for this component. This may save some conversions.

Reply:
@chapulina, I tried to be coherent with the other CMD commands. 

 - AngularVelocityCmd -> math::Vector3d
 - JointForceCmd -> std::vector<double>
 - JointVelocityCmd -> std::vector<double>
 - LinearVelocityCmd -> math::Vector3D
 - PoseCmd -> math::Pose3d

none of them are using the message type

Reply:
The commands aren't using SDF types either :wink: We have a couple of components using messages to hold their data, but it's true that there are more using SDF. I think this is mainly historical though, we started thinking we'd use SDF more widely, but I think that's becoming less true. See #494 for more context.

The reason I suggested a message is that it matches better what the components do. They hold data and are serialized. Right now, using SDF, you're still relying on a message for serialization. If you just used the message directly in the type, then you save the SDF <-> msgs conversion by using a `MsgSerializer` directly.